inductiva/resources/machine_groups.py

```diff
@@ -118,6 +118,30 @@ def list(fout: TextIO = sys.stdout):
         print("No active computational resources found.", file=fout, end="")
 
 
+def get_by_name(machine_name: str):
+    """Returns the machine group corresponding to `machine_name`."""
+    machine_groups = _fetch_machine_groups_from_api()
+    machine_group = next(
+        (mg for mg in machine_groups if mg["name"] == machine_name), None)
+    if machine_group is not None:
+        mg_class = _get_machine_group_class_from_api_response(machine_group)
+        return mg_class.from_api_response(machine_group)
+    return None
+
+
+def _get_machine_group_class_from_api_response(mg: dict):
```

Comment:
make it more explicit. It is easier for testing purposes and decouples it from having to have a dict.
What if the dict doesn't have the required kesy?

```suggestion
def _get_machine_group_class_from_api_response(type: str, is_elastic: bool):
    """Get the suitable machine group class for the given arguments""" 
```